Build the macOS port of dhewm3 1.5.5 against SDL3.

SDLMain.m is Objective-C, so the extern "C" declaration of SDL_main does not
compile; SDL3 renamed SDL_QUIT to SDL_EVENT_QUIT. Apply in neo/ with patch -p0.

--- sys/osx/SDLMain.m
+++ sys/osx/SDLMain.m
@@ -15,7 +15,8 @@
   // as SDL.h doesn't implicitly include SDL_main.h anymore,
   // declare SDL_main() here. I think it's the only part of SDL_main.h we used,
   // we implement it in DOOMController.mm an call it here in applicationDidFinishLaunching
-  extern "C" int SDL_main( int argc, char *argv[] );
+  // (this is an Objective-C file: no extern "C", it already has C linkage)
+  extern int SDL_main( int argc, char *argv[] );
 #else // SDL2 and SDL1.2
   #include "SDL.h"
 #endif
@@ -88,7 +89,12 @@
 {
     /* Post a SDL_QUIT event */
     SDL_Event event;
+    SDL_zero(event);
+#ifdef D3_SDL3
+    event.type = SDL_EVENT_QUIT;
+#else
     event.type = SDL_QUIT;
+#endif
     SDL_PushEvent(&event);
 }
 @end
